python/export_util.py

```diff
@@ -235,3 +311,287 @@ def csv_query(
+
+
+def write_keys(
+    graph: List[Union[Node, Relationship]],
+    output: io.StringIO,
+    config: mgp.Map,
+):
+    node_keys = dict()
+    rel_keys = dict()
+    for element in graph:
+        if element.get("type") == "node":
+            if element.get("labels"):
+                if config.get("format").upper() == "TINKERPOP":
+                    node_keys.update({"labelV": get_type_string("labelV")})
+                else:
+                    node_keys.update(
+                        {"labels": get_type_string("labels")}
+                    )  # what if two nodes have same property name but different value type? or node's property name is "labels"?  # noqa: E501
+
+            for key, value in element.get("properties").items():
+                node_keys.update({key: get_type_string(value)})
+
+        elif element.get("type") == "relationship":
+            if config.get("format").upper() == "TINKERPOP":
+                rel_keys.update({"labelE": get_type_string("labelE")})
+            else:
+                rel_keys.update({"label": get_type_string("label")})
+
+            for key, value in element.get("properties").items():
+                rel_keys.update({key: get_type_string(value)})
+
+    if config.get("format").upper() == "GEPHI":
+        node_keys.update({"TYPE": get_type_string("TYPE")})
+        rel_keys.update({"TYPE": get_type_string("TYPE")})
+
+    for key, value in node_keys.items():
+        output.write(f'<key id="{key}" for="node" attr.name="{key}"')
+        if config.get("useTypes"):
+            if isinstance(value, str):
+                output.write(f' attr.type="{value}"')
+            else:
+                output.write(f' attr.type="{value[0]}" attr.list="{value[1]}"')
+        output.write("/>\n")
+    for key, value in rel_keys.items():
+        output.write(f'<key id="{key}" for="edge" attr.name="{key}"')
+        if config.get("useTypes"):
+            if isinstance(value, str):
+                output.write(f' attr.type="{value}"')
+            else:
+                output.write(f' attr.type="{value[0]}" attr.list="{value[1]}"')
+        output.write("/>\n")
+
+
+def write_graph(output: io.StringIO):
+    output.write('<graph id="G" edgedefault="directed">\n')
+
+
+def get_gephi_label_value(
+    element: Union[Node, Relationship], config: mgp.Map
+) -> str:
+    for caption in config.get("caption"):
+        if caption in element.get("properties").keys():
+            return str(element.get("properties").get(caption))
+
+    if element.get("properties").values():
+        return str(list(element.get("properties").values())[0])
+
+    return str(element.get("id"))
+
+
+def write_labels_as_data(
+    element: Union[Node, Relationship], output: io.StringIO, config: mgp.Map
+):
+    if not element.get("labels"):
+        return
+    if config.get("format").upper() == "TINKERPOP":
+        output.write('<data key="labelV">')
+        for i in range(0, len(element.get("labels"))):
```

Comment:
Check here how to write this with enumerate: https://realpython.com/python-enumerate/